src/DeFuncto.Core/Types/Result.cs

```diff
@@ -114,6 +114,18 @@ public Unit Iter(Action<TOk> iteratorOk, Action<TError> iteratorError)
         [Pure]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator Result<TOk, TError>(TError error) => Error(error);
+
+        public override bool Equals(object obj)
+        {
+            var nval = obj as Result<TOk, TError>?;
+            return nval != null && Equals((Result<TOk, TError>)nval);
+        }
+
+        public bool Equals(Result<TOk, TError> other) =>
+            IsOk == other.IsOk && other.value.Equals(value);
```

Comment:
The IsOk is implicitly tested in the equals of the `Du`, just test the equality of the DU.